colosseum/constants.py

```diff
@@ -1,72 +1,45 @@
-from . import parser
-from . import units
+from .validators import (is_color, is_integer, is_length, is_number,
+                         is_percentage)
 
 
 class Choices:
-    "A class to define allowable data types for a property"
+    "A class to define allowable data types for a property."
+
     def __init__(
-            self, *constants, length=False, percentage=False,
-            integer=False, number=False, color=False,
+            self, *constants, validators=None,
             explicit_defaulting_constants=None):
         self.constants = set(constants)
         self.explicit_defaulting_constants = explicit_defaulting_constants or []
-        self.length = length
-        self.percentage = percentage
-        self.integer = integer
-        self.number = number
-        self.color = color
+        self.validators = validators or []
 
     def validate(self, value):
-        if self.length or self.percentage:
-            try:
-                val = parser.units(value)
-                if self.length:
-                    return val
-                elif self.percentage and isinstance(val, units.Percent):
-                    return val
-            except ValueError:
-                pass
-        if self.integer:
-            try:
-                return int(value)
-            except (ValueError, TypeError):
-                pass
-        if self.number:
-            try:
-                return float(value)
-            except (ValueError, TypeError):
-                pass
-        if self.color:
-            try:
-                return parser.color(value)
-            except ValueError:
-                pass
+        for validator in self.validators:
+            error_msg, value = validator(value)
+            if not error_msg:
+                return value
```

Comment:
Any reason to do this as a "return tuple, check for empty result", rather than declaring a ValidationError exception class, and catching the exception if there's a problem?

Reply:
Hmmm not really, can update this :-)